composer/datasets/in_context_learning_evaluation.py

```diff
@@ -675,3 +638,145 @@ def get_icl_task_dataloader(
         get_num_samples_in_batch=dataset.get_num_samples_in_batch,
         split_batch=dataset.split_batch if isinstance(dataset, InContextLearningMultipleChoiceTaskDataset) else None,
     )
+
+
+def partition_dataset_by_category(dataset_uri: str, destination_path: str) -> Dict[str, str]:
+    """If has_categories is enabled, we partition the dataset into a separate dataset for each category value in the data and writes each partition to a local file.
+
+    Args:
+        dataset_uri (str): Location of dataset.
+        destination_path (str): Base destination path, we will write a separate partition off this URI for each category.
+
+    Raises:
+        MissingConditionalImportError: If datasets not installed raise exception.
+        Exception: If 'category' key missing from dataset, raise exception.
+    Returns:
+        Dict[str, str]: Mapping of category names to partitioned dataset local files names.
+    """
+    try:
+        from datasets import load_dataset  # pyright: ignore [reportGeneralTypeIssues]
+    except ImportError as e:
+        raise MissingConditionalImportError(extra_deps_group='nlp',
+                                            conda_package='datasets',
+                                            conda_channel='conda-forge') from e
+    with dist.local_rank_zero_download_and_wait(destination_path):
+        if dist.get_local_rank() == 0:
+            get_file(dataset_uri, destination_path, overwrite=True)
+    dataset = load_dataset('json', data_files=destination_path, split='train', streaming=False)
+    if 'category' not in dataset.features.keys():
+        raise Exception(
+            f"Attempted to partition dataset by category but it doesn't have a category key: {str(dataset)}")
```

Comment:
what does `str(dataset)` print? can you double check this works and/or just use a string here? like maybe the `dataset_uri` would be good?